bridges/otelzap/zap_encoder.go

```diff
@@ -0,0 +1,233 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) 2016-2017 Uber Technologies, Inc.
+
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+// this file implements object and array encoder - similar to memory encoder by zapcore.
+var (
+	_ zapcore.ObjectEncoder = (*ObjectEncoder)(nil)
+	_ zapcore.ArrayEncoder  = (*ArrayEncoder)(nil)
+)
+
+// Object Encoder implements zapcore.ObjectEncoder.
+// It encodes given fields to OTel attribute.
+type ObjectEncoder struct {
```

Comment:
Does it need to be exported?

```suggestion
type objectEncoder struct {
```